### ISSUE TITLE
[DOCS] validation and scoring on same test set

### ISSUE THREAD

**Issue by aneesh-joshi** [keras-team/keras#8796]
This is my first issue. I hope this is the right place for it.

In the keras mnist cnn example : https://github.com/keras-team/keras/blob/master/examples/mnist_cnn.py

the validation and testing uses the same set `x_test`

This might give of the wrong idea to some of the new comers that test sets can be used for validation.
If this is a valid problem, I will make a PR in which I split the train set into train and validate.

```
model.fit(x_train, y_train,
          batch_size=batch_size,
          epochs=epochs,
          verbose=1,
          validation_data=(x_test, y_test))

score = model.evaluate(x_test, y_test, verbose=0)
```

- [ x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [ x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by vianamp** [comment]
Hi. To validate on a subset of the training data, you can simple use the argument `validation_split` when you can model.fit(...).

**Comment by aneesh-joshi** [comment]
Hi @vianamp , I made the changes with validation_split... in the PR: https://github.com/keras-team/keras/pull/8812

Hopefully, it's worthy of being merged.

**Comment by aneesh-joshi** [comment]
as per @fchollet , this isn't an issue.
Closing the issue.
As per him:
> Both ways are valid, but show different functionality.
> 
> The purpose of the examples is just to demonstrate Keras functionality. Meanwhile, the purpose of using a separate validation set is to avoid "validation set overfitting", i.e. avoid making model-tuning choices that leak validation set information into the final model. This is not a problem with this example since the architecture choices are arbitrary, rather than tuned for performance on the test set.
> Thus, we won't merge this change.
